core/src/main/java/com/amplitude/core/Configuration.kt

```diff
@@ -8,21 +8,21 @@ import com.amplitude.core.utilities.InMemoryStorageProvider
 typealias EventCallBack = (BaseEvent, status: Int, message: String) -> Unit
 
 open class Configuration @JvmOverloads constructor(
-    val apiKey: String,
-    var flushQueueSize: Int = FLUSH_QUEUE_SIZE,
-    var flushIntervalMillis: Int = FLUSH_INTERVAL_MILLIS,
-    var instanceName: String = DEFAULT_INSTANCE,
-    var optOut: Boolean = false,
-    val storageProvider: StorageProvider = InMemoryStorageProvider(),
```

Comment:
Should this be va r or val? do we allowed to override?

Reply:
Good point, I was wondering about this as well. That time I changed to var to enable it to be set, taking another thought, maybe we can use this as a val first until it is need in the future. Let me change it back to val for now.